stage1/init/cgroup.go

```diff
@@ -219,9 +238,17 @@ func createCgroups(root string, machineID string, appHashes []types.Hash) error
 		}
 		subcgroup = filepath.Join(slicePath, unit, "system.slice")
 	} else {
-		escapedmID := strings.Replace(machineID, "-", "\\x2d", -1)
-		machineDir := "machine-" + escapedmID + ".scope"
-		subcgroup = filepath.Join("machine.slice", machineDir, "system.slice")
+		if machinedRegister() {
+			escapedmID := strings.Replace(machineID, "-", "\\x2d", -1)
+			machineDir := "machine-" + escapedmID + ".scope"
+			subcgroup = filepath.Join("machine.slice", machineDir, "system.slice")
+		} else {
```

Comment:
I wonder if it worth it or if we should just disable the feature when !fromUnit && !machinedRegister